lib/entryList.ts

```diff
@@ -0,0 +1,50 @@
+import { Channel } from './channel'
+import {
+  EntryListAPI,
+  OnEntryListUpdatedHandler,
+  OnEntryListUpdatedHandlerProps,
+  OnEntryListUpdatedHandlerReturn,
+} from './types'
+
+const isFunction = (f: any) => typeof f === 'function'
+
+const runHandler = async (
+  handler: OnEntryListUpdatedHandler | null,
+  handlerArg: OnEntryListUpdatedHandlerProps
+) => {
+  try {
+    // Handler was not registered. Registering a handler is required.
+    if (!handler) {
+      throw new Error('Registering an onEntryListUpdated handler is required')
+    }
+
+    // await will accept both async and sync functions, no need for the isPromise check
+    return await handler(handlerArg)
+  } catch (error) {
+    console.error(error)
+    return false
+  }
+}
+
+export default function createEntryList(channel: Channel): EntryListAPI {
+  let _handler: OnEntryListUpdatedHandler | null = null
+
+  channel.addHandler(
+    'entryList',
```

Comment:
Shall we call this `entryListUpdated` in case there are more events in the future?
With this name, it looks like this is the only event that can happen in entry location.